### PR TITLE
chore(bug): be more verbose (-x) and fail if error code...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

be more verbose (-x) and fail if error code > 0 occurs (-e) when doing docker buildx or arbitrary-users-patch scripts

Change-Id: I2519add533d0fa6fc8f4c1f1c121a4fdf867b5ae
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nickboldt** [comment]
Followup to https://github.com/eclipse-che/che-devfile-registry/pull/362 so we don't have "everything is happy" mattermost notifications when container builds fail :)

**Review by ericwill — APPROVED** [review]

File: .github/workflows/nightly-build-publish.yml [hunk 1]
```diff
@@ -59,10 +59,10 @@ jobs:
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
 
       - name: Build and push base images
-        run: /bin/bash arbitrary-users-patch/build_images.sh --push
+        run: /bin/bash -xe arbitrary-users-patch/build_images.sh --push
```
Comment (benoitf) [inline .github/workflows/nightly-build-publish.yml:62]: Shouldn't we add flags in the script ?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -76,6 +76,7 @@ jobs:
           git config --global user.email "mkuznets@redhat.com"
 
           export GITHUB_TOKEN=${{ secrets.CHE_BOT_GITHUB_TOKEN }}
+          set -xe
```
Comment (benoitf) [inline .github/workflows/release.yml:79]: if flag is set in the Bash script we won't need it there

File: .github/workflows/nightly-build-publish.yml [hunk 3]
```diff
@@ -59,10 +59,10 @@ jobs:
...
 
       - name: Build and push happy path image
-        run: /bin/bash arbitrary-users-patch/happy-path/build_happy_path_image.sh --push
+        run: /bin/bash -xe arbitrary-users-patch/happy-path/build_happy_path_image.sh --push
```
Comment (benoitf) [inline .github/workflows/nightly-build-publish.yml:65]: Also why do we need to use Bash prefix and not ./arbitrary-users-patch/happy-path/build_happy_path_image.sh

File: .github/workflows/nightly-build-publish.yml (hunk 1 above)
Comment (nickboldt) [reply .github/workflows/nightly-build-publish.yml:62]: that's an option, yes. Wasn't sure if there was an intentional reason for having the script not fail for failures. If we want the -e insidethe script, I'm good with that.

File: .github/workflows/release.yml (hunk 2 above)
Comment (nickboldt) [reply .github/workflows/release.yml:79]: true... again, not sure if we want to fail for everything that could go wrong in make-release (like all the error code 1 crap that `git` throws for benign warnings). So... yeah perhaps it's not needed here.

File: .github/workflows/nightly-build-publish.yml (hunk 1 above)
Comment (benoitf) [reply .github/workflows/nightly-build-publish.yml:62]: I don't see a case where we want to continue on failure inside the script

File: .github/workflows/release.yml (hunk 2 above)
Comment (benoitf) [reply .github/workflows/release.yml:79]: but we should only call git commands if there is something to do, not expect it should fail and ignore error for some case

File: .github/workflows/nightly-build-publish.yml (hunk 1 above)
Comment (nickboldt) [reply .github/workflows/nightly-build-publish.yml]: arbitrary-users-patch scripts already set -e so I guess this is redundant.

File: .github/workflows/nightly-build-publish.yml (hunk 3 above)
Comment (nickboldt) [reply .github/workflows/nightly-build-publish.yml]: no idea... that's how it was. But I agree, that's weird.

**Review by benoitf — APPROVED** [review]

**Pull request merged**